src/websocket.js

```diff
@@ -94,39 +99,67 @@ class WebSocket extends EventTarget {
   }
 
   get onopen() {
-    return this.listeners.open;
+    return this.onopenCb;
   }
 
   get onmessage() {
-    return this.listeners.message;
+    return this.onmessageCb;
   }
 
   get onclose() {
-    return this.listeners.close;
+    return this.oncloseCb;
   }
 
   get onerror() {
-    return this.listeners.error;
+    return this.onerrorCb;
   }
 
   set onopen(listener) {
-    delete this.listeners.open;
-    this.addEventListener('open', listener);
+    if (listener === null) {
+      if (this.onopenCb !== null) {
+        this.removeEventListener('open', this.onopenCb);
+      }
+    } else {
+      this.onopen = null;
```

Comment:
infinity recursion

Reply:
It's not infinite recursion at all. If the callback is null, it never takes that branch